js/lib/bandstructure.js

```diff
@@ -784,6 +796,38 @@ BandPlot.prototype.updateDosPlot = function (orientation = 'vertical') {
     bandPlotObject.dosSeries = [];
     curve = [];
 
+
+    // Here, a dumb dataset was created to represent the y axis.
+    // All the DOS curves were filled to this dumb dataset.
+    var dosx = bandPlotObject.dosData['dos'][0]['x'];
+    var dosy = bandPlotObject.dosData['dos'][0]['y'];
+
+    dosx.forEach(function (data, k) {
```

Comment:
Add a comment here to briefly explain why it can fill the background up to fermi energy.
Is the curve.push a special plot function that will draw the line rather than the background?

Reply:
Yes, I add it in the code. Basically, if we only give the dataset up to the Fermi level, the fillcolor will only show for the whole curve range.